rootfs/deploy.py

```diff
@@ -0,0 +1,49 @@
+import docker
+import os
+import tarfile
+import os.path
+import boto3
+import requests
+
+from botocore.client import Config
+from urllib.parse import urlparse
+
+
+def log_output(stream):
+    for chunk in stream:
+        print(chunk)
+
+
+if os.getenv('TAR_URL'):
+    if os.path.exists("/var/run/secrets/object/store/access-key-id"):
+        with open("/var/run/secrets/object/store/access-key-id", encoding='utf-8') as the_file:
+            secretId = the_file.read().strip()
+        with open("/var/run/secrets/object/store/access-secret-key", encoding='utf-8') as the_file:
+            secretKey = the_file.read().strip()
+        url = urlparse(os.getenv('TAR_URL'))
+        hosturl = url.scheme + "://" + url.netloc
+        s3 = boto3.client('s3', config=Config(signature_version='s3v4'),
+                          endpoint_url=hosturl,
+                          aws_access_key_id=secretId,
+                          aws_secret_access_key=secretKey)
+        bucket = url.path.split('/')[1]
+        key = ('/').join(url.path.split('/')[2:])
+        s3.download_file(bucket, key, 'apptar')
+    else:
+        r = requests.get(os.getenv('TAR_URL'))
+        with open("apptar", "wb") as app:
+            app.write(r.content)
+
+print ("download tar file complete")
```

Comment:
The space before the parens should be removed--`print` is just another function. Same for lines 40 and 47.